### PR TITLE
Finished Employee Controller and Test

### PR DESCRIPTION
**Pull request by nikx1015**
# Description

Created an employee controller which allows the user to add, edit, and view employees. I also created a test to make sure the controller will allow the user to retrieve a single employee, all employees, add a new employee, and edit an employee.

## Related Ticket(s)
Ticket 6
Allow devs to access employee resource

## Steps to Test
git add and git commit what you're working on on your branch
git checkout master to switch to your local master branch
git fetch --all will give you access to your teammate's branches (if they've pushed them up)
git checkout na-employee to access my branch.
If you don't see the most recent version of their files, try git pull origin na-employee
Run your server on their branch and make sure everything looks okay
Once server is running, use your browser to navigate to api/employees to ensure that correct information is displayed.

Afterwards, in Visual Studio, go to the EmployeeTest.cs file under TestBangazonAPI and right-click on it and click on Run Tests. Ensure that all tests come back green.

Check verbiage and comments to make sure they are legible and understandable.

# Checklist:
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by ColonelSped** [comment]
Cannot run in current form. Check names of Computer

File: BangazonAPI/BangazonAPI/Controllers/EmployeesController.cs [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+                        };
+                        //assigning computer to employee id
+                        if (!reader.IsDBNull(reader.GetOrdinal("Computer Id")))
+                        {
+                            Computer computer = new Computer()
+                            {
+                                id = reader.GetInt32(reader.GetOrdinal("Computer Id")),
+                                make = reader.GetString(reader.GetOrdinal("make")),
+                                manufacturer = reader.GetString(reader.GetOrdinal("manufacturer")),
+                                datePurchased = reader.GetDateTime(reader.GetOrdinal("purchaseDate")),
```
Comment (ColonelSped) [inline BangazonAPI/BangazonAPI/Controllers/EmployeesController.cs]: datePurchased doesn't work here.

**Comment by nikx1015** [comment]
Made changes to names for Computer based on updated Computer model.

File: BangazonAPI/BangazonAPI/Controllers/EmployeesController.cs [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+                            };
+                            employee.employeeComputer = computer;
+                        }
+
+                        //checking for decommission date
+
+                        if (!reader.IsDBNull(reader.GetOrdinal("decomissionDate")))
+                        {
+                            employee.employeeComputer.dateDecommissioned = reader.GetDateTime(reader.GetOrdinal("decomissionDate"));
```
Comment (ColonelSped) [inline BangazonAPI/BangazonAPI/Controllers/EmployeesController.cs]: dateDecomissioned doesn't work here.

**Review by ColonelSped — CHANGES_REQUESTED** [review]

**Review by ColonelSped — APPROVED** [review]

**Comment by ColonelSped** [comment]
Based on Ticket #6 you have added user functionality that isn't needed. User should not be able to Delete and the Test doesn't need Delete either. You can create a super secret deletion method for creating and deleting in your database for testing purposes. 

**_User should be able to GET a list, and GET a single item.
The employee's department name should be included in the employee representation

A representation of the computer that the employee is currently using should be included in the employee representation

Testing Criteria
Write a testing class and test methods that validate the GET single, GET all, POST, and PUT operations work as expected._**

**Comment by ColonelSped** [comment]
Looks good! 👍

**Comment by BatsGoScreechh** [comment]
Looks good to me! 👍

**Pull request merged**